package/boot/uboot-mvebu/Makefile

```diff
@@ -66,6 +66,11 @@ define Package/u-boot/install
 	$(if $(findstring cortexa53,$(BUILD_SUBTARGET)),,$(Package/u-boot/install/default))
 endef
 
+define Build/Configure
+	$(SED) '/CONFIG_WATCHDOG_AUTOSTART/d' $(PKG_BUILD_DIR)/configs/$(UBOOT_CONFIG)_defconfig
```

Comment:
Based on your commit message, this is only required for Marvell ESPRESSObin, but now it is going to be enabled to all boards in mvebu. I just want to avoid another situation like this one https://github.com/openwrt/openwrt/pull/9798/. Wouldn't it be better to create a patch, which I did in the past, to enable it only to the relevant board and perhaps to send it upstream?

Reply:
For mvebu, it's only disabled for this one board. But yeah, it might be a trap in the future, fixed ;)

I'm not sure about upstream, I think it makes sense to keep it disabled by default, because it always depends on what distro you're booting. And here we're pretty distro specific and can be sure that the upcoming stable release has the watchdog timer enabled.

Reply:
I agree with @BKPepe, this should be a separate patch to U-boot defconfig and sent upstream.

Reply:
Okay, switched to a patch instead of the sed line.
I'd at least wait until final 22.03 is released before sending this upstream though.

Reply:
Right, but I still think it's too early for that. Once a release is out doesn't mean that users update instantly, it'll take time.
I'll send the oneliner in the future.

Reply:
Sorry, but this makes no sense to me.
What is the reason to wait until sending this upstream?

Reply:
Isn't it obvious? If you run a distro that doesn't have the watchdog driver and update the firmware where uboot enables it, the device will boot loop forever, basically rendering it useless for the user until she/he fixes it manually.  The current OpenWrt stable release is such a release.

If you can expect the majority of users to have updated to a distro that has the driver, it makes sense to flip the default for upstream, but not before.

Only the next OpenWrt release will have that driver. It's not yet released. Neither will everybody update instantly after release. It'll take some time. So I'll wait a few month to send that patch. Very similar to how one deprecates stuff. You don't throw it out instantly, you mark it as deprecated and throw it out in a year or even further down the line.

Reply:
Yeah, this isn't u-boot as part of the distro partition or something, the firmware gets flashed to NOR.
It's common to download a firmware binary and just flash that, especially since it's such a PITA to build this firmware by hand.
I run this a firmware build here but with a Debian release. The next guy may run Arch's firmware with OpenWrt stable.

Reply:
I get what your intention is, but this will not prevent people from randomly flashing whatever they find.
You gotta look at it from OpenWrt-s angle, why should we keep patches and modifications downstream?

Reply:
It's not about this oneliner, but the fact that this kind of custom stuff just keeps on pilling in U-boot and eventually nobody knows why its even here, seen it too many times.

Reply:
Yeah, I know what you mean and I actually agree in general. But still, there're cases where patches make sense here but not upstream.
Anyway, I removed that commit from this PR, so let's consider that as resolved ;)